### PR TITLE
Ik 100

### PR DESCRIPTION
**Pull request by handriamanantena**
Inbox no longer crashes when a user selects the inbox tab.

### CODE REVIEW

File: Backend/getUserInboxList.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+$result = mysqli_query($con, $sql);
+
+$routeIds = array();
+while($row = mysql_fetch_array($result))
+{
+    #querry to get all driver ids from routes. 
+    $sql= "SELECT user_id FROM Routes_Users_Association WHERE route"
+
+}
+if(!$result){
```
Comment (SaifMahabub) [inline Backend/getUserInboxList.php]: Control statement "if" should have one space between the control keyword and opening parenthesis according to PHP conventions.

File: Backend/getUserInboxList.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			(SELECT route_id FROM (SELECT * FROM Routes_Users_Association) AS copy WHERE user_id LIKE (SELECT id FROM Users WHERE email like '".$driver_email."')),
+			(SELECT id FROM Users WHERE email LIKE '".$passenger_email."'))";
+
+#querry to get all pending routes from user
+$sql = "SELECT route_id FROM Routes_Users_Association WHERE user_id LIKE (SELECT id FROM Users WHERE email like '".$passenger_email."') AND user_confirm_route =0"
+
+$result = mysqli_query($con, $sql);
+
+$routeIds = array();
+while($row = mysql_fetch_array($result))
```
Comment (SaifMahabub) [inline Backend/getUserInboxList.php]: Control statement "while" should have one space between the control keyword and opening parenthesis according to PHP conventions.

File: client/app/src/main/java/com/example/mrides/Activity/InboxActivity.java [hunk 3]
```diff
@@ -47,13 +56,38 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        mRecyclerView.setAdapter(mAdapter);
-        if(getIntent().getParcelableExtra("NOTIFICATION")!=null) {
-            notification = (RemoteMessage) (getIntent().getParcelableExtra("NOTIFICATION"));
-            ((InboxAdapter) mAdapter).setViewComponents(notification.getData());
-        }
+        getInboxData(); //TODO call this method to get the list of inbox data
+
+    }
 
+    private void getInboxData(){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Activity/InboxActivity.java]: Space missing before opening curly bracket.

File: client/app/src/main/java/com/example/mrides/Activity/InboxActivity.java [hunk 4]
```diff
@@ -47,13 +56,38 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
+    @Override
+    public void Update(String response) {
+        System.out.println("Inbox list.:"+ response);
+        handlepopulateInboxResponse(response);
+        mAdapter = new InboxAdapter(this,invitations);
+        mRecyclerView.setAdapter(mAdapter);
+    }
+
+    private void handlepopulateInboxResponse(String response){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Activity/InboxActivity.java]: Space missing before opening curly bracket.

File: client/app/src/main/java/com/example/mrides/Activity/InboxActivity.java [hunk 5]
```diff
@@ -47,13 +56,38 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        try {
+            JSONArray invites = new JSONArray(response);
+            for(int index =0;index<invites.length();index++){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Activity/InboxActivity.java]: Space missing before opening curly bracket and after keyword "for"

File: client/app/src/main/java/com/example/mrides/Activity/MainActivity.java [hunk 6]
```diff
@@ -170,6 +170,13 @@ private void firebaseAuthWithGoogle(GoogleSignInAccount acct) {
     public void Update(String response) {
 
         requestHandler.detach(this);
+        System.out.println("Response: "+ response);
+        if(response.contains("User")){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java]: Space missing before opening curly bracket and after keyword "if"

File: client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java [hunk 7]
```diff
@@ -27,20 +37,14 @@
...
 
-    private Map<String,String> responseBody;
-    private Context inboxContext;
-    private Dialog dialog;
-    private RequestHandler requestHandler = new RequestHandler();
+    private List<Invitation> invitations = new ArrayList<>();
+    protected static Context inboxContext;
 
-    public InboxAdapter(Context inboxContext){
+    public InboxAdapter(Context inboxContext,List<Invitation> invitations){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java]: Space missing before opening curly bracket

File: client/app/src/main/java/com/example/mrides/userDomain/DriverJSONMap.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
+package com.example.mrides.userDomain;
+
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DriverJSONMap {
+
+    public static Map<String, String> getParameters(User driver){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/userDomain/DriverJSONMap.java]: Space missing before opening curly bracket

File: client/app/src/main/java/com/example/mrides/userDomain/PassengerSerializer.java [hunk 9]
```diff
@@ -19,14 +19,19 @@
      */
     //TODO the paramter needs to be changed to a passenger in the future. For now user works
     public static Map<String, String> getParameters(User user){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/userDomain/PassengerSerializer.java]: Space missing before opening curly bracket and after keyword "for"

File: client/app/src/main/java/com/example/mrides/Activity/HomePage.java [hunk 10]
```diff
@@ -197,6 +197,7 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                 startActivity(intentSettings);
                 break;
             case R.id.nav_logout:
+                System.out.println("INSIDE");
```
Comment (adrielFab) [inline client/app/src/main/java/com/example/mrides/Activity/HomePage.java]: if this is not needed any longer for debugging, remove it

File: client/app/src/main/java/com/example/mrides/Activity/MainActivity.java [hunk 11]
```diff
@@ -195,6 +202,7 @@ public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
             // User is signed in
             user = new User(firebaseuser,googleuser);
             requestHandler.setUser(user);
+            requestHandler.attach(this);
             requestHandler.httpPostStringRequest("http://"+getString(R.string.web_server_ip)+"/register_user.php",
                     UserSerializer.getParameters(user), "application/x-www-form-urlencoded; charset=UTF-8",
```
Comment (adrielFab) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java]: the content type should be set to a CONSTANT

File: client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java [hunk 12]
```diff
@@ -87,90 +92,35 @@ public void onBindViewHolder(ViewHolder holder, int position) {
...
-    }
 
     /**
      * Provide a reference to the views for each data item
      * Complex data items may need more than one view per item, and
      * you provide access to all the views for a data item in a view holder
      */
 
-    public static class ViewHolder extends RecyclerView.ViewHolder {
+    public class ViewHolder extends RecyclerView.ViewHolder{
```
Comment (adrielFab) [inline client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java]: missing space before curly bracket

File: client/app/src/main/java/com/example/mrides/Notification/RegistrationTokenInstanceIdService.java [hunk 13]
```diff
@@ -13,20 +17,30 @@
  * https://firebase.google.com/docs/cloud-messaging/android/first-message
  */
 
-public class RegistrationTokenInstanceIdService extends FirebaseInstanceIdService {
+public class RegistrationTokenInstanceIdService extends FirebaseInstanceIdService implements ActivityObserver{
```
Comment (adrielFab) [inline client/app/src/main/java/com/example/mrides/Notification/RegistrationTokenInstanceIdService.java]: missing space

File: client/app/src/main/java/com/example/mrides/Notification/UserProfileListener.java [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+    /**
+     * This method creates the user profile diologue
+     */
+    private void createDiolgue() {
+        dialog = new Dialog(inboxContext);
+        dialog.setTitle(inboxContext.getString(R.string.invited_to_route));
+        dialog.setContentView(R.layout.userprofile_dialog_layout);
+        dialog.show();
+
+        TextView textViewFullName = (TextView) dialog.findViewById(R.id.textViewFirstName);
```
Comment (adrielFab) [inline client/app/src/main/java/com/example/mrides/Notification/UserProfileListener.java:59]: Implement these fields with attribute visibility

**Pull request merged**